### PR TITLE
merge compute_output_stat

### PR DESCRIPTION
**Pull request by njzjz**

### CODE REVIEW

**Comment by wanghan-iapcm** [comment]
> I don't know how to compute for dipole and polar - TF has no such method.

I would propose compute the avg (like the atom_bias_e) and std of label. By far we remove the avg, it is also possible to scale the output of the fitting net by the std. 

The polar of the TF uses the statistics of the diagonal of the avg.

**Comment by wanghan-iapcm** [comment]
I realize that the std for each type is not available. We may need to uniformly scale the atomic output based on statistics.

File: deepmd/utils/out_stat.py
[the remark this answers is not in the document]
Comment (wanghan-iapcm) [reply deepmd/utils/out_stat.py:1]: We may have to consider two cases: 

1. only "global" labels are available. We can compute the bias for each type by lstsq. 
2. atomic labels are available. We can compute the bias and std for each atom type. 

please see the tf polar fitting as an example.

**Review by wanghan-iapcm — CHANGES_REQUESTED** [review]
please check my comments on [deepmd/utils/out_stat.py](https://github.com/deepmodeling/deepmd-kit/pull/3310/files/4ba3ae76195eade16d6e2861eef7dbd7cd6b6a64#diff-f5e155304c725fd1a236003a3245699bf295a45a59cc5af14236fd2837cd7c9b)

File: deepmd/utils/out_stat.py
[the remark this answers is not in the document]
Comment (njzjz) [reply deepmd/utils/out_stat.py:1]: I am confused about tf polar fitting. It seems it does `compute_output_stats` things in the `compute_input_stats` method.

https://github.com/deepmodeling/deepmd-kit/blob/d629616305f8754e8a4115dc244a7057bd2d6b1e/deepmd/tf/fit/polar.py#L135-L136

[the remark this answers is not in the document]
Comment (wanghan-iapcm) [reply deepmd/utils/out_stat.py:1]: interesting... It it a bug.

File: source/tests/common/test_out_stat.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+            rtol=1e-7,
+        )
+
+    def test_compute_stats_from_atomic(self):
+        bias, std = compute_stats_from_atomic(self.output, self.atype)
+        np.testing.assert_allclose(bias, self.mean)
+        np.testing.assert_allclose(
+            std,
+            self.std,
+            rtol=1e-2,
```
Comment (wanghan-iapcm) [inline source/tests/common/test_out_stat.py:80]: we may manually compute the reference std from data... the tol basically cannot detect issue.

**Review by wanghan-iapcm — CHANGES_REQUESTED** [review]

File: source/tests/common/test_out_stat.py (hunk 1 above)
Comment (njzjz) [reply source/tests/common/test_out_stat.py:80]: okay... It seems the output is fixed when I set the random seed.

**Review by wanghan-iapcm — APPROVED** [review]

File: source/tests/common/test_out_stat.py (hunk 1 above)
Comment (wanghan-iapcm) [reply source/tests/common/test_out_stat.py]: or you may want to mock the random generator

**Pull request merged**